docs/design.md

```diff
@@ -0,0 +1,115 @@
+Design notes
+============
+
+Motivation
+----------
+
+To run GitHub Actions self-hosted runners faster and stably by making full use of
+the machine resource in the staging environment.
+
+Goals
+-----
+
+- Deploy and manage GitHub Actions self-hosted runners on multiple bare-metal
+  servers easily with Kubernetes.
+- Enable runners to finish a time-consuming initialization step before jobs are
+  assigned in order not to make users wait longer.
+- Extend lifetime of runners from outside when jobs are failed, to investigate
+  what causes the failure.
+- Notify users whether jobs are failed or not via Slack and extend the lifetime
+  from Slack.
```

Comment:
Can you have a pluggable architecture that allows us to use notifications other than slack?

Reply:
Currently No, but this architecture is simple and easy to extend, so that might be possible in the future.